services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py

```diff
@@ -19,3 +24,62 @@ async def validate_project_ownership(
         != user_id
     ):
         raise ProjectInvalidRightsError(user_id=user_id, project_uuid=project_uuid)
+
+
+async def get_user_project_access_rights(
+    app: web.Application,
+    project_id: ProjectID,
+    user_id: UserID,
+    product_name: ProductName,
+) -> UserProjectAccessRights:
+    """
+    This function resolves user access rights on the project resource.
+
+    If project belong to user personal workspace (workspace_id = None) then it is resolved
+    via user <--> groups <--> projects_to_groups.
+
+    If project belong to shared workspace (workspace_id not None) then it is resolved
+    via user <--> groups <--> workspace_access_rights
+    """
+    db: ProjectDBAPI = app[APP_PROJECT_DBAPI]
+
+    project_db = await db.get_project_db(project_id)
+    if project_db.workspace_id:
+        workspace = await get_workspace(
+            app,
+            user_id=user_id,
+            workspace_id=project_db.workspace_id,
+            product_name=product_name,
+        )
+        _user_project_access_rights = UserProjectAccessRights(
+            uid=user_id,
+            read=workspace.my_access_rights.read,
+            write=workspace.my_access_rights.write,
+            delete=workspace.my_access_rights.delete,
+        )
+    else:
+        _user_project_access_rights = (
+            await db.get_pure_project_access_rights_without_workspace(
+                user_id, project_id
+            )
+        )
+    return _user_project_access_rights
+
+
+async def has_user_project_access_rights(
+    app: web.Application,
+    *,
+    project_id: ProjectID,
+    user_id: UserID,
+    permission: PermissionStr,
```

Comment:
minor: Ideally it would probably make sense to have an enum here.

Reply:
I second this